### PR TITLE
feat: rework tx table to be more responsive and show transfers

### PR DESCRIPTION
**Pull request by reallybeard**
## Description

The transaction rows were a little tricky in their current implementation to adapt to mobile sizes, so I re-worked the way they are built and used the resizeObserver to actually adjust the row based on its container's width. This gives us very fine grained control over how it looks at different sizes.

The other thing that was added was displaying the actual TX transfers, this makes it much easier on us to show correct information and the parsing was getting pretty tricky to figure out who and what got the assets.

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

No issue

## Risk

Medium Risk

I did modify the way these work, the data is all there just the way its rendered is different.

Another note, we are using the resizeObserver so any performance degradation is possible here. This won't prevent people from seeing the information but it may slow down their browser.

## Testing

View the TX table at all screen sizes, and in all the places you can view them /tx-history, asset details page, dashboard sidebar.

Note down any slow down in performances so we can try and troubleshoot.

## Screenshots
Screen Shot 2022-03-25 at 9 10 10 AM
Screen Shot 2022-03-25 at 9 11 25 AM
Screen Shot 2022-03-25 at 12 12 20 PM

### CODE REVIEW

**Comment by NeOMakinG** [comment]
Awesome.

I guess performance will be degraded only when you resize the browser and `txRows` changes right?

I'm afraid the biggest performance issue would be when you switch from portrait to landscape mode on mobile, some old mobiles could burn 😄

**Comment by reallybeard** [comment]
> Awesome.
> 
> I guess performance will be degraded only when you resize the browser and `txRows` changes right?
> 
> I'm afraid the biggest performance issue would be when you switch from portrait to landscape mode on mobile, some old mobiles could burn 😄

Lol yah I mean really switch from portrait to landscape wouldn't be that bad, its an instant width change. Vs dragging the window size around. But testing this on the deployment seems to be very snappy on resize.

File: src/components/TransactionHistoryRows/TransactionGenericRow.tsx [hunk 1]
```diff
@@ -69,94 +63,124 @@ export const TransactionGenericRow = ({
...
+    <Button
+      height='auto'
+      fontWeight='inherit'
+      variant='unstyled'
+      w='full'
+      p={4}
+      onClick={() => toggleOpen()}
+    >
+      <SimpleGrid
+        gridTemplateColumns={{ base: '1fr', md: columns }}
```
Comment (GMSteuart) [inline src/components/TransactionHistoryRows/TransactionGenericRow.tsx:96]: If you pass the columns in as an array, chakra will use that value for each of the breakpoints in the theme starting from 0 to largest breakpoint allowing you to remove the checks above.

```tsx
gridTemplateColumns={{ base: '1fr', md: ['1fr 2fr', '1fr 2fr', '1fr 2fr 1fr 1fr'] }}
```

Comment (reallybeard) [reply src/components/TransactionHistoryRows/TransactionGenericRow.tsx:96]: i'm not trying to use the viewport size, but the size of the parent container width.

Comment (reallybeard) [reply src/components/TransactionHistoryRows/TransactionGenericRow.tsx:96]: The checks above are for the size of the parent container, those media queries will use the viewport size.

Comment (reallybeard) [reply src/components/TransactionHistoryRows/TransactionGenericRow.tsx:96]: Sure would be cool if chakra let you override the source of the breakpoint, very edge case but would be neat.

**Comment by reallybeard** [comment]
> Noticed something that is not a blocker, but not sure if this is expected. The trade has the status showing first while sends/receives/yearn txs have the TXID showing first
> 
> Screen_Shot_2022-03-26_at_7 38 11_AM
> 
> But aside from this, this looks good to me. Checked tx on dashboard, assets/accounts, and transaction history. Tried in mobile as well.

Good catch, I think the status makes the most sense after the txid I’ll move it.

**Review by 0xdef1cafe — APPROVED** [review]

**Review by gomesalexandre — APPROVED** [review]
Tested locally and looks good, status order is also consistent across Tx types.

**Pull request merged**